### PR TITLE
Build: Add network stack drivers into build

### PR DESCRIPTION
**Pull request by savvamitrofanov**
This commit adds basic set of NetworkPkg drivers with HTTP boot support.
Also adds missing DxePcdLib into HiiDatabaseDxe.

### CODE REVIEW

File: Docs/Sample.plist [hunk 1]
```diff
@@ -1611,6 +1611,150 @@
 				<key>Path</key>
 				<string>Ext4Dxe.efi</string>
 			</dict>
+			<dict>
```
Comment (PMheart) [inline Docs/Sample.plist:1614]: In my opinion, only put these drivers in SampleCustom.plist.

File: build_oc.tool [hunk 2]
```diff
@@ -158,11 +158,20 @@ package() {
     cp "${arch}/Shell.efi" "${dstdir}/${arch}/EFI/OC/Tools/OpenShell.efi" || exit 1
 
     efiDrivers=(
+      "ArpDxe.efi"
```
Comment (PMheart) [inline build_oc.tool:161]: Instead of alphabetical order, shall we classify by authors (e.g. ours vs external)?

File: OpenCorePkg.dsc [hunk 3]
```diff
@@ -167,7 +184,10 @@
   MdeModulePkg/Bus/Isa/Ps2MouseDxe/Ps2MouseDxe.inf
   MdeModulePkg/Bus/Isa/Ps2KeyboardDxe/Ps2KeyboardDxe.inf
   MdeModulePkg/Bus/Usb/UsbMouseDxe/UsbMouseDxe.inf
-  MdeModulePkg/Universal/HiiDatabaseDxe/HiiDatabaseDxe.inf
+  MdeModulePkg/Universal/HiiDatabaseDxe/HiiDatabaseDxe.inf {
+    <LibraryClasses>
+      PcdLib|MdePkg/Library/DxePcdLib/DxePcdLib.inf
```
Comment (vit9696) [inline OpenCorePkg.dsc:189]: DxePcdLib requires pcd protocols, which legacy firmware may not have.

File: Docs/Sample.plist (hunk 1 above)
Comment (savvamitrofanov) [reply Docs/Sample.plist:1614]: Yes, I agree, this is not a common set of drivers, I'll remove them from Sample.plist

File: build_oc.tool (hunk 2 above)
Comment (savvamitrofanov) [reply build_oc.tool:162]: I think this will complicate maintaining this in the future, so I think we should leave it as it is

**Pull request merged**